R/brreg-utils.R

```diff
@@ -0,0 +1,87 @@
+#' null2na
+#' @noRd
+null2na <- function(x) if (is.null(x) || length(x) == 0) NA else x
+
+#' request_brreg
+#' @noRd
+request_brreg <- function(type = c('enheter', 'kommuner'), entity = NULL) {
+  type <- match.arg(type)
+  req <- request(brreg_url)
+  if (type == 'enheter') {
+    is_valid_number <- grepl('\\d{9}', entity)
+    #
+    if (nchar(entity) != 9 && !grepl('\\D', entity) ) { #grepl('\\d', entity) && grepl('\\D', entity)
```

Comment:
Perform mod11 check on organisation number to check if its valid. The function below could work as a solution to this. Also, instead of checking for a number, use regex to extract only numbers, ie. use `gsub()` instead of `grepl()`.

Test for valid organisation numbers:

```r
mod11 <- \(x, n = 8, weights = c(3,2,7,6,5,4,3,2)) {
  x <- trimws(as.character(x))
  x <- as.integer(unlist(strsplit(x, '')))
  k <- sum(weights * x[1:n]) %% 11
  if (k == 0) k == x[n+1] else (11 - k) == x[n+1]
}
```

Reply:
Good idea. As discussed I think we should separate the actual mod11 check, `sum(weights * x[1:n]) %% 11`), from the cleaning of organisation numbers, particularly since we are attempting to handling both numbers and names in the same function. But I'll take a look at both.

Reply:
Plese review the combined input checks `clean_entity_input()`, `verify_entity_number()` and `mod11()`